api/audit/tasks.py

```diff
@@ -73,3 +79,52 @@ def create_audit_log_from_historical_record(
         master_api_key=history_instance.master_api_key,
         **instance.get_extra_audit_log_kwargs(history_instance),
     )
+
+
+@register_task_handler()
+def create_segment_priorities_changed_audit_log(
+    previous_id_priority_pairs: typing.List[typing.Tuple[int, int]],
+    feature_segment_ids: typing.List[int],
+    user_id: int = None,
+    master_api_key_id: int = None,
+):
+    """
+    This needs to be a separate task called by the view itself. This is because the OrderedModelBase class
+    that the FeatureSegment model inherits from implicitly uses bulk update to move other feature segments
+    out of the way.
+
+    Given the implementation, it's not really possible (or perhaps desirable) to determine which moves
+    were actually made by the user. The logic is such that it iterates over the list of priorities sent
+    by the client. Take for example 2 overrides:
+
+     segment_a - priority 0
+     segment_b - priority 1
+
+    A change to move segment_b to priority 0 will set the priority on segment_b to 0 (triggering a single
+    write to the database). segment_a will be moved from 0 -> 1 using bulk update. This seems fine as we
+    then know b was moved and don't really care that a was moved out of the way. If the user, however moves
+    segment_a to priority 1, the single save will be done on segment_b since it'll be first in the list and
+    segment_a will be bulk updated.
+
+    This is a very simple case, and it likely gets more complicated / difficult to follow if there are more
+    than 2 overrides.
+    """
+
+    # TODO: use previous priorities to show what changed.
```

Comment:
I will do this in a separate PR alongside the rest of the changes to include information about what changed.